File: api/playwright/python/typeText.js
const { spawn } = require('child_process');
const path = require('path');

class DesktopAutomator {
    constructor() {
        this.pythonScript = path.join(__dirname, 'typeText.py');
    }
    
    async typeText(text, options = {}) {
        return new Promise((resolve, reject) => {
            const optionsJson = JSON.stringify(options);
            const args = [this.pythonScript, text, optionsJson];
            
            const python = spawn('python', args);
            let output = '';
            let error = '';
            
            python.stdout.on('data', (data) => {
                output += data.toString();
            });
            
            python.stderr.on('data', (data) => {
                error += data.toString();
            });
            
            python.on('close', (code) => {
                if (code !== 0) {
                    reject(new Error(`Python script failed: ${error}`));
                    return;
                }
                
                try {
                    const result = JSON.parse(output.trim());
                    if (result.status === 'success') {
                        resolve(result);
                    } else {
                        reject(new Error(result.message));
                    }
                } catch (parseError) {
                    reject(new Error(`Failed to parse Python output: ${output}`));
                }
            });
        });
    }
}

// Usage
async function main() {
    const automator = new DesktopAutomator();
    
    try {
        // Simple usage
        await automator.typeText("Hello my World");
        console.log('First automation completed');
        
        // With options
        // await automator.typeText("New text", {
        //     delay: 0.2,
        //     clearFirst: true
        // });
        // console.log('Second automation completed');
        
    } catch (error) {
        console.error('Automation failed:', error);
    }
}

main();

File: api/playwright/python/typeText.py
import pyautogui
import sys
import time
import json

def type_text(text, options=None):
    if options is None:
        options = {}
    
    delay = options.get('delay', 0.1)
    clear_first = options.get('clearFirst', False)
    
    try:
        # Disable pyautogui failsafe for automation
        pyautogui.FAILSAFE = False
                
        # Wait for focus
        time.sleep(delay)
        
        # Clear existing text if requested
        if clear_first:
            pyautogui.hotkey('ctrl', 'a')
            time.sleep(0.05)
        
        # Type text
        pyautogui.typewrite(text)
        
        return {"status": "success", "message": "Automation completed"}
        
    except Exception as e:
        return {"status": "error", "message": str(e)}

if __name__ == "__main__":
    if len(sys.argv) < 2:
        result = {"status": "error", "message": "Usage: python typeText.py <text> [options_json]"}
    else:
        text = sys.argv[1]
        options = json.loads(sys.argv[2]) if len(sys.argv) > 2 else {}
        
        result = type_text(text, options)
    
    print(json.dumps(result))